freqtrade/configuration/configuration.py

```diff
@@ -170,9 +168,19 @@ def _process_common_options(self, config: Dict[str, Any]) -> None:
 
     def _process_datadir_options(self, config: Dict[str, Any]) -> None:
         """
-        Extract information for sys.argv and load datadir configuration:
-        the --datadir option
+        Extract information for sys.argv and load directory configurations
+        --user-data, --datadir
         """
+        if 'user_data_dir' in self.args and self.args.user_data_dir:
+            config.update({'user_data_dir': self.args.user_data_dir})
+        elif 'user_data_dir' not in config:
+            # Default to cwd/user_data (legacy option ...)
+            config.update({'user_data_dir': str(Path.cwd() / "user_data")})
+
+        # reset to user_data_dir so this contains the absolute path.
+        config['user_data_dir'] = create_userdata_dir(config['user_data_dir'])
```

Comment:
Ah, a more serious note: the user dir with its layout (subdirs) should not be created just because I mentioned it in the cli option. You can check its existence here, but creation should only be made by explicit call to the create-userdir command...

Reply:
I am not certain about this - we might be facing "quite a few" problems if we change `user_data` to something else - since this has always been there.

You think we should fail startup if a folder is not present while checking and hae the user issue the `create-userdir` command explicitly?

won't users think "just create that folder" ?

Reply:
No, they won't,.

Exactly, yes -- the bot should check for existence of the dir and quit if it cannot find it. 

The dir path can simply be misspelled, it should not be created just after first run of the bot, the `create-userdir` command should be issued by the user explicitly. Same for the subdirs needed for the running mode. A user can find she misspelled the dir, change it and make another error in the name -- another dir will be created. This will quickly get the environment to complete mess...

So the behavior, imho, should be following: check for dir, same for necessary subdirs (can be done in the modules, not in the main startup configuration, but in the configuration as well). If run `create-userdir` -- create userdir and its subdirs. Take into account that subdir can point to another locations (overriden with corresponding `--*-dir` option or `*_dir` setting in the config,

Reply:
> Take into account that subdir can point to another locations (overriden with corresponding --*-dir option or *_dir setting in the config),

which is something i would want to get rid off since there's no point IMHO.

Yes, modules create the dir if they need it (plot, for example).
I would not want to remove the dir-creation from all modules unless we remove the override possibility, and restrict users to the --user-dir directory.
Otherwise, we'd need to have a "create-userdir" - "create-plotdir", "create-...dir" command - which will become a complete mess and make 0 sense in my opinion, and will only create frustrations.